src/Parsers/parseQuery.cpp

```diff
@@ -260,6 +260,9 @@ ASTPtr tryParseQuery(
     Expected expected;
     ASTPtr res;
     const bool parse_res = parser.parse(token_iterator, res, expected);
+    // If statement is perfect match parser's last token, the ending semicolon is
+    // not touched. Manual touch it.
+    token_iterator.get();
```

Comment:
Why do other queries with `ON CLUSTER` clause not need this to work?

Reply:
Other queries with `ON CLUSTER` clause not need this to work because the `ON CLUSTER` token is not last token parser. For example last token parse `ParserCreateFunctionQuery` is `lambda_p`. In this situation last token parser `lambda_p` will touch the ending semicolon;
I’ve posted more details on issue #45303 . Not only `ON CLUSTER` clause cause this problem. Other statement may also cause this problem. Please refer if you have a bit of time:)

Reply:
Can we modify `ParserDropFunctionQuery` instead?

Reply:
> Can we modify ParserDropFunctionQuery instead?

No, because the root cause is not related to DROP or ON CLUSTER

Reply:
> > Can we modify ParserDropFunctionQuery instead?
> 
> No, because the root cause is not related to DROP or ON CLUSTER

yeah, you got it!

Reply:
Any update? Any addition work should be done?